HFStrategy/Strategy/PositionManager.py

```diff
@@ -0,0 +1,236 @@
+import logging
+from enum import Enum
+from .Position import Position
+from ..utils.CustomLogger import CustomLogger
+
+# Simple wrapper to log the calling of a function
+# to enable set the logger to debug mode
+def logfunc(func):
+    def wrapper(*args, **kwargs):
+      args[0].logger.debug("['{0}'] params: {1} kwargs: {2}".
+                           format(func.__name__, args, kwargs))
+      return func(*args, **kwargs)
+    return wrapper
+
+class PositionError(Exception):
+  ''' 
+  An Error that is thrown whenever there is a problem with opening,
+  creating or updating a position.
+  '''
+  def __init__(self, message, errors=None):
+        # Pass the message to the base class
+        super().__init__(message)
+        self.errors = errors
+
+class OrderType(Enum):
+  '''
+  An Enum used to represent the diffrent types of orders that 
+  are possible.
+  '''
+  MARKET = 1
+  EXCHNAGE_MARKET = 2
+  LIMIT = 3
+  EXCHNAGE_LIMIT = 4
+
+class PositionManager(object):
+
+  ############################
+  # Close Position functions #
+  ############################
+
+  @logfunc
+  def closePosition(self, *args, **kwargs):
+    self.closePositionWithOrder(*args, **kwargs)
+  
+  @logfunc
+  def closeOpenPositions(self):
+    openPositions = list(self.positions.values())
+    count = len(openPositions)
+    for pos in openPositions:
+      def close_pos(self):
+        price, mts = self.getLastPrice(pos.symbol)
+        self.closePositionMarket(
+            symbol=pos.symbol, price=price, mtsCreate=mts, tag='Close all positions')
+      self._startNewThread(close_pos)

```

Comment:
Why do you start a new thread here? `closePositionMarket` calls through to `closePositionWithOrder` which submits the order on a new thread, it looks like 2 threads are spawned here.